crypto/ed25519/private_test.go

```diff
@@ -146,3 +146,49 @@ func TestPrivateKey_Kind(t *testing.T) {
 		})
 	}
 }
+
+func TestPrivateKey_Sign(t *testing.T) {
+	tests := []struct {
+		name    string
+		pk      PrivateKey
+		msg     []byte
+		want    []byte
+		wantErr bool
+	}{
+		{
+			"success-charlotte",
+			charlottePrivateKey,
+			[]byte("message"),
+			[]byte("}Q\xea\xfaRx1i\xd0\xa9J\x0c\x9f+\xca\xd5\xe0=)\x173\x00\x93\x0f\xf3\xc7\xd6;\xfdd\x17\xae\x1b\xc8\x1f\xefQ\xba\x14\x9a\xe8\xa1\xe1\xda\xe0_\xdc\xa5\x07\x8b\x14\xba\xc4\xcf&\xcc\xc6\x01\x1e^\xabw\x03\x0c"),
+			false,
+		},
+		{
+			"success-sofia",
+			sofiaPrivateKey,
+			[]byte("egassem"),
+			[]byte("\xdel\x88\xe6\x9c\x9f\x93\x0bY\xdd\xf4\x80\xc2\x9aUy\xec\x89\\\xa9z6\xf6it\xc1\xf0\x15\\\xc0fu.\xcd\x9a\x9bA5\xd2r2\xe0T\x80\xbc\x98X\x01\xa9\xfd\xe4'\xc7\xef\xa5B_\x0fFI\xb8\xad\xbd\x05"),
+			false,
+		},
+		{
+			"err-len",
+			PrivateKey{
+				key: []byte{0xd, 0x9b, 0x4a, 0x3c, 0x10, 0x72, 0x19, 0x91, 0xc6, 0xb8, 0x6, 0xf0, 0xf3, 0x43, 0x53, 0x5d, 0xc2, 0xb4, 0x6c, 0x74},
+			},
+			[]byte("message"),
+			nil,
+			true,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			got, err := tt.pk.Sign(tt.msg)
+			if (err != nil) != tt.wantErr {
+				t.Errorf("Sign() error = %v, wantErr %v", err, tt.wantErr)
+				return
+			}
+			if !reflect.DeepEqual(got, tt.want) {
```

Comment:
You can then switch this line for `if !assert.Equal(tt.want, got) {` notice the `got` and `tt.want` switched places. Now the failure messages are easy to work with.